modin/core/storage_formats/pandas/aggregations.py

```diff
@@ -0,0 +1,588 @@
+        sums_of_squares = (df**2).sum().rename(MODIN_UNNAMED_SERIES_LABEL)
+        count = pandas.Series(
+            np.repeat(len(df), len(df.columns)), index=df.columns, copy=False
+        ).rename(MODIN_UNNAMED_SERIES_LABEL)
+        return sums, sums_of_squares, count
+
+    @staticmethod
+    def _compute_nan_aggs(
+        raw_df: np.ndarray, cols: pandas.Index, nan_mask: np.ndarray
+    ) -> Tuple[pandas.DataFrame, pandas.DataFrame, pandas.DataFrame]:
+        """
+        Compute sums, sums of square and the number of observations for a partition assuming there are NaN values in it.
+
+        Parameters
+        ----------
+        raw_df : np.ndarray
+            Raw values of the partition to compute the aggregations for.
+        cols : pandas.Index
+            Columns of the partition.
+        nan_mask : np.ndarray[bool]
+            Boolean mask showing positions of NaN values in the `raw_df`.
+
+        Returns
+        -------
+        Tuple[sums: pandas.DataFrame, sums_of_squares: pandas.DataFrame, count: pandas.DataFrame]
+            A tuple storing DataFrames where each of them holds the result for
+            one of the described aggregations.
+        """
+        # Unfortunately, in case of NaN values we forced to compute multiple sums/square sums/counts
+        # for each column because we have to exclude values at positions of NaN values in each other
+        # column individually.
+        # Imagine we have a dataframe like this:
+        #   col1: 1, 2  , 3  , 4
+        #   col2: 2, NaN, 3  , 4
+        #   col3: 4, 5  , NaN, 7
+        # In this case we would need to compute 2 different sums/square sums/count for 'col1':
+        #   - The first one excluding the values at the NaN possitions of 'col2' (1 + 3 + 4)
+        #   - And the second one excluding the values at the NaN positions of 'col3' (1 + 2 + 4)
+        # and then also do the same for the rest columns. At the end this should form a matrix
+        # of pairwise sums/square sums/counts:
+        #   sums[col1, col2] = sum(col1[i] for i in non_NA_indices_of_col2)
+        #   sums[col2, col1] = sum(col2[i] for i in non_NA_indices_of_col1)
+        #   ...
+        # Note that sums[col1, col2] != sums[col2, col1]
+        sums = {}
+        sums_of_squares = {}
+        count = {}
+
+        # TODO: is it possible to get rid of this for-loop somehow?
+        for i, col in enumerate(cols):
+            # Here we're taking each column, resizing it to the original frame's shape to compute
+            # aggregations for each other column and then excluding NaN values by setting zeros
+            # using the validity mask:
+            #  col1: 1, 2  , 3  , 4   df[i].resize()  col1: 1, 2, 3, 4  putmask()  col1: 1, 2, 3, 4
+            #  col2: 2, NaN, 3  , 4   ------------->  col1: 1, 2, 3, 4  -------->  col1: 1, 0, 3, 4
+            #  col3: 4, 5  , NaN, 7                   col1: 1, 2, 3, 4             col1: 1, 2, 0, 4
```

Comment:
`excluding NaN value` and the dataframe with nans in the example is misleading. It can be clarified that all nans have already been replaced by zeros. And that it is not nanos that are excluded here, but the positions where these nanos were for the correct calculation of the `sums`, `sums_of_squares`, and `count`.
```suggestion
            # aggregations for each other column and then excluding NaN values by setting zeros
            # using the validity mask:
            #  col1: 1, 2  , 3  , 4   df[i].resize()  col1: 1, 2, 3, 4  putmask()  col1: 1, 2, 3, 4
            #  col2: 2, NaN, 3  , 4   ------------->  col1: 1, 2, 3, 4  -------->  col1: 1, 0, 3, 4
            #  col3: 4, 5  , NaN, 7                   col1: 1, 2, 3, 4             col1: 1, 2, 0, 4
```

Reply:
Rephrased the comment, hope it's clearer now

Reply:
thanks